src/docs/implementations/dynatrace.adoc

```diff
@@ -4,25 +4,190 @@ Jon Schneider <jschneider@pivotal.io>
+It can ingest multi-purpose dimensional time-series data and has built-in dashboarding. Both SaaS and self-hosted (Managed) deployments are offered.
 
 include::install.adoc[]
 
 == Configuring
 
+For setting up new integrations with Dynatrace, it is recommended to use the latest version of the https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v2/[Dynatrace Metrics API] (`api-version: v2`).
+More details on how to use the v2 API are provided further <<bookmark-apiv2, below>>.
+
+
+=== Using a Dynatrace OneAgent installed on the host (preferred) [[bookmark-oneagent-example]]
+
+If a Dynatrace OneAgent is installed on the host running Micrometer, metrics can be exported directly using the OneAgent without having to specify an endpoint URI or API token.
+Use the following code in your project to export Micrometer metrics to the https://www.dynatrace.com/support/help/how-to-use-dynatrace/metrics/metric-ingestion/ingestion-methods/local-api/[local OneAgent endpoint]:
+
 [source,java]
 ----
 DynatraceConfig dynatraceConfig = new DynatraceConfig() {
+    @Override
+    public DynatraceApiVersion apiVersion() {
+        return DynatraceApiVersion.V2;
+    }
+
+    @Override
+    @Nullable
+    public String get(String k) {
+        // This method for retrieving arbitrary config items introduced by the interface is currently not used in DynatraceConfig.
+        return null;
+    }
+};
+MeterRegistry registry = new DynatraceMeterRegistry(dynatraceConfig, Clock.SYSTEM);
+----
+
+If you are working with Spring Boot, you can instead put the following properties in your `application-dynatrace.yml` in the `resources` folder.
+
+[source,yml]
+----
+management.metrics.export.dynatrace:
+    enabled: true
+    api-version: v2
+----
+
+=== Using a custom endpoint
+
+If no Dynatrace OneAgent is available, both the Dynatrace Metrics API v2 endpoint and an API token have to be specified.
+The https://www.dynatrace.com/support/help/dynatrace-api/basics/dynatrace-api-authentication/[Dynatrace API token documentation] contains more information on how to create an API token.
+The 'Ingest metrics' (`metrics.ingest`) permission is required on the token in order to ingest metrics.
+It is recommended to limit scope to only this permission.
+
+[source,java]
+----
+DynatraceConfig dynatraceConfig = new DynatraceConfig() {
+    @Override
+    public DynatraceApiVersion apiVersion() {
+        return DynatraceApiVersion.V2;
+    }
+
+    @Override
+    public String uri() {
+        // The endpoint of the Dynatrace Metrics API v2 including path, e. g.
+        // "https://{your-environment-id}.live.dynatrace.com/api/v2/metrics/ingest".
+        // To be read from environment variables or a config file.
+        return MY_DYNATRACE_URI;
+    }
+
     @Override
     public String apiToken() {
+        // It is *not* recommended to place secrets in config files but rather to read them from env vars, for example.
```

Comment:
I would just say "read them from a secure source". Environment variables are not necessarily secure/best solution, though they can be ok in certain situations.
There are other occurrences in this doc, could you please fix those too?